src/Symfony/EventListener/QueryParameterValidateListener.php

```diff
@@ -85,6 +71,20 @@ public function onKernelRequest(RequestEvent $event): void
             return;
         }
 
+        if ($operation && $this->provider instanceof ProviderInterface) {
+            if (null === $operation->getQueryParameterValidationEnabled()) {
+                $operation = $operation->withQueryParameterValidationEnabled($request->isMethodSafe() && 'GET' === $request->getMethod() && $operation instanceof CollectionOperationInterface);
+            }
+
+            $this->provider->provide($operation, $request->attributes->get('_api_uri_variables') ?? [], [
+                'request' => $request,
+                'uri_variables' => $request->attributes->get('_api_uri_variables') ?? [],
+                'resource_class' => $operation->getClass(),
+            ]);
+
+            return;
+        }
```

Comment:
can you add a non reg test in QueryParameterValidateListenerTest

Reply:
Tests added. I did some changes in addition on moved code.  
Could you check if the if block stay good ?

Reply:
I have an error on CI that I don't have in my local env. I work on it this day.

Reply:
Pipeline is ok.
The problem with phpunit tests was the namespace. I didn't see that test class was already exist in Symfony sources.  
For my future contributions, why there are tests in ./tests/Symfony folder and others in ./src/Symfony/Tests ?

Reply:
We moved some tests into their components directory on the main branch, part of our subtree-split work. See: https://github.com/api-platform/core/blob/main/CONTRIBUTING.md#components-tests